packages/next/next-server/server/next-server.ts

```diff
@@ -565,9 +616,8 @@ export default class Server {
     this.handleCompression(req, res)
 
     try {
-      const fn = this.router.match(req, res, parsedUrl)
-      if (fn) {
-        await fn()
+      const matched = await this.router.execute(req, res, parsedUrl)
+      if (matched) {
```

Comment:
This should always have matched since we a final catchall route right?

Reply:
True but just in case it did not match we want to still show a 404 page. Eg when the routes config is changed.